src/main/java/blf/core/writers/XesWriter.java

```diff
@@ -174,38 +161,32 @@ public void addStringList(@NonNull String key, @NonNull List<String> values) {
     }
 
     @Override
-    protected void writeState(String filenameSuffix) throws Throwable {
+    protected void writeState(String fileNameSuffix) {
```

Comment:
Unfortunately, your logic changes in this function were what we had to change for the streaming mode and the safe batching mode to work. I believe that you have to revert them in order for them to work again.